src/audio/tdfb/tdfb.c

```diff
@@ -358,6 +523,49 @@ static int tdfb_cmd_set_data(struct comp_dev *dev,
 	return ret;
 }
 
+static int tdfb_cmd_enum_set(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_data *cd)
+{
+	int ret = 0;
+
+	switch (cdata->index) {
+	case CTRL_INDEX_DIRECTION:
+		if (cdata->num_elems) {
+			cd->az_value = cdata->chanv[0].value;
```

Comment:
we should sanity check userdata, otherwise once fuzzer finds this it is going to have a pile of fun breaking it

Reply:
Yep, it's still missing. I'll add that tomorrow into next version of PR.

Reply:
I just updated the PR. I added check for num_elems and value. For switch I didn't add value check since kernel should limit it to 0 and 1. The choice for angle value limit was tricky. The topology compiler accepts a low number (16?) but in kernel and FW there's no hard limit so I ended up to use 360 as max. One degree angle step would be nicer than current 30.